benchmarks/bench.sh

```diff
@@ -524,7 +529,19 @@ run_imdb() {
     $CARGO_COMMAND --bin imdb -- benchmark datafusion --iterations 5 --path "${IMDB_DIR}" --prefer_hash_join "${PREFER_HASH_JOIN}" --format parquet -o "${RESULTS_FILE}"
 }
 
+# Runs the external aggregation benchmark
+run_external_aggr() {
+    # Use TPC-H SF1 dataset
+    TPCH_DIR="${DATA_DIR}/tpch_sf1"
+    RESULTS_FILE="${RESULTS_DIR}/external_aggr.json"
+    echo "RESULTS_FILE: ${RESULTS_FILE}"
+    echo "Running external aggregation benchmark..."
 
+    # Only parquet is supported
+    # External aggregation is not stable yet, set partitions to 4 to make sure
```

Comment:
what isn't stable about it? Like it uses too many file handles or something?

Reply:
I have updated the comment to be more specific, just to ensure this benchmark can run successfully on different machines
```
    # Since per-operator memory limit is calculated as (total-memory-limit / 
    # number-of-partitions), and by default `--partitions` is set to number of
    # CPU cores, we set a constant number of partitions to prevent this 
    # benchmark to fail on some machines.
```
Moreover, now memory-limited queries fail quite easily due to `record_batch.get_array_memory_size()` over-count the memory usage https://github.com/apache/datafusion/issues/13089, once it's fixed we can set default partitions larger